io_u.c

```diff
@@ -895,7 +915,13 @@ static int fill_io_u(struct thread_data *td, struct io_u *io_u)
 		goto out;
 
 	set_rw_ddir(td, io_u);
-
+	/*
+	 * check if we have received Invalid Direction
+	 */
+	if (io_u->ddir == DDIR_INVAL) {
+		dprint(FD_IO, "Invalid Direction received ddir = %d", io_u->ddir);
+		return 1;
```

Comment:
@axboe by doing this we bail out of the do_io() while loop (https://github.com/axboe/fio/blob/5090d1d0f2a109c276384c93308566b7a3bfa5ad/backend.c#L997 ) and won't have done `fio_mark_td_terminate()` (see https://github.com/axboe/fio/blob/5090d1d0f2a109c276384c93308566b7a3bfa5ad/backend.c#L965 ). Basically the intent is to stop us processing the io_u because this diff changes things so we return earlier than we would have done according to the rate we were trying to achieve which means the io_u is sent earlier and skews the numbers. However it's become a bit messy...